File: TsinghuaNetHelper/ConnectHelper.cpp
#include "pch.h"

#include "ConnectHelper.h"

using namespace std;
using namespace winrt;

namespace winrt::TsinghuaNetHelper::implementation
{
    template <typename... Args>
    IConnect GetHelperT(NetState const& state, Args&&... args)
    {
        switch (state)
        {
        case NetState::Net:
            return NetHelper(forward<Args>(args)...);
        case NetState::Auth4:
            return Auth4Helper(forward<Args>(args)...);
        case NetState::Auth6:
            return Auth6Helper(forward<Args>(args)...);
        default:
            return nullptr;
        }
    }

    IConnect ConnectHelper::GetHelper(NetState const& state)
    {
        return GetHelperT(state);
    }

    IConnect ConnectHelper::GetHelper(NetState const& state, hstring const& username, hstring const& password)
    {
        return GetHelperT(state, username, password);
    }
} // namespace winrt::TsinghuaNetHelper::implementation

File: TsinghuaNetUWP.Helper/ConnectHelper.vb
Imports Berrysoft.Tsinghua.Net

Public Module ConnectHelper
    Public Function GetHelper(state As NetState) As IConnect
        Select Case state
            Case NetState.Net
                Return New NetHelper()
            Case NetState.Auth4
                Return New Auth4Helper()
            Case NetState.Auth6
                Return New Auth6Helper()
            Case Else
                Return Nothing
        End Select
    End Function

    Public Function GetHelper(state As NetState, username As String, password As String) As IConnect
        Select Case state
            Case NetState.Net
                Return New NetHelper(username, password)
            Case NetState.Auth4
                Return New Auth4Helper(username, password)
            Case NetState.Auth6
                Return New Auth6Helper(username, password)
            Case Else
                Return Nothing
        End Select
    End Function
End Module
